### PR TITLE
IT-5075 | RadioButton and Checkbox

### PR DESCRIPTION
**Pull request by tweetgeek**

### CODE REVIEW

File: lib/js/components/Form/Toggle/Toggle.vue [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+			'icon': $color-neutral-icon-disabled,
+		),
+		'selected': (
+			'label': $color-neutral-text-heavy-disabled,
+			'icon': $color-primary-icon-disabled,
+		),
+	),
+	'loading': (
+		'not-selected': (
+			'label': $color-neutral-text-heavy-disabled,
```
Comment (kvas-damian) [inline lib/js/components/Form/Toggle/Toggle.vue:67]: loading should like `default`, not `disabled`: https://miro.com/app/board/o9J_knPnexA=/?moveToWidget=3458764546392448081&cot=14

File: lib/js/components/Form/SelectionControl/SelectionControl.vue [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+	&__labelWrapper {
+		display: flex;
+	}
+
+	&__icon {
+		position: relative;
+	}
+
+	&.-x-small {
+		#{$root}__label {
```
Comment (kvas-damian) [inline lib/js/components/Form/SelectionControl/SelectionControl.vue:229]: label isn't correctly positioned for xSmall size

File: lib/js/components/Form/SelectionControl/SelectionControl.vue [hunk 3]
```diff
@@ -0,0 +1,352 @@
+<template>
+	<div
```
Comment (kvas-damian) [inline lib/js/components/Form/SelectionControl/SelectionControl.vue:2]: most likely, if you convert this `div` into a `label` then you could remove `click` handler. Clicks on a `label` tag automagically toggles checkbox/radio

File: lib/js/components/Form/SelectionControl/SelectionControl.vue [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+	@each $name, $map in $selection-control-colors {
+		@if map-get($map, 'hoverable') {
+			&.-#{$name} {
+				@each $size, $map in $selection-control-sizes {
+					&.-#{$size} {
+						#{$root}__icon::before {
+							bottom: -#{map-get($map, 'iconBorder')};
+							left: -#{map-get($map, 'iconBorder')};
+							right: -#{map-get($map, 'iconBorder')};
+							top: -#{map-get($map, 'iconBorder')};
```
Comment (kvas-damian) [inline lib/js/components/Form/SelectionControl/SelectionControl.vue:106]: it looks more like an outline size. Maybe `iconOutlinePadding`?

File: lib/js/components/Form/SelectionControl/SelectionControl.vue [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+					background-color: transparent;
+					border-radius: 50%;
+					content: '';
+					position: absolute;
+					transform: scale(0);
+					transition: all $default-transition-time ease 0ms;
+					z-index: -1;
+				}
+
+				&:hover,
```
Comment (kvas-damian) [inline lib/js/components/Form/SelectionControl/SelectionControl.vue:123]: I'm not convinced that defining these styles in for loop makes it easier to understand. What would you say about moving it into regular styles block

File: lib/js/components/Form/SelectionControl/SelectionControl.vue [hunk 6]
```diff
@@ -0,0 +1,352 @@
...
+		@click="onToggle"
+	>
+		<div class="selectionControl__iconWrapper">
+			<icon :icon="icon" :size="iconSize" class="selectionControl__icon" />
+		</div>
+		<div v-if="label" class="selectionControl__labelWrapper">
+			<div class="selectionControl__label">{{ label }}</div>
+		</div>
+		<input
+			type="checkbox"
```
Comment (kvas-damian) [inline lib/js/components/Form/SelectionControl/SelectionControl.vue:23]: should it be `checkbox` when we use it to render radio?

File: lib/js/components/Form/SelectionControl/SelectionControl.vue (hunk 3 above)
Comment (tweetgeek) [reply lib/js/components/Form/SelectionControl/SelectionControl.vue:2]: but it is not good practice to use div in label

Comment (kvas-damian) [reply lib/js/components/Form/SelectionControl/SelectionControl.vue:2]: why don't you use `spans` inside label?

File: lib/js/components/Form/SelectionControl/SelectionControl.vue (hunk 2 above)
Comment (kvas-damian) [reply lib/js/components/Form/SelectionControl/SelectionControl.vue:182]: not checkbox/radio is centered when label goes into multiline

**Comment by kvas-damian** [comment]
> I'm wondering if we should implement something similar to `v-radio-group` https://vuetifyjs.com/en/components/radio-buttons/ to make sure that logic which deselects other selected radios is in one place and tested well.
> 
> Also we could think about improving component API by adopting features from vuetify: https://vuetifyjs.com/en/components/checkboxes/#model-as-array and https://vuetifyjs.com/en/components/checkboxes/#model-as-boolean

we agreed offline, that we can do that we can implement it in future

File: lib/js/components/Form/SelectionControl/SelectionControl.vue [hunk 7]
```diff
@@ -0,0 +1,339 @@
...
+			<icon :icon="icon" :size="iconSize" class="selectionControl__icon" />
+		</div>
+		<div v-if="label" class="selectionControl__labelWrapper">
+			<div class="selectionControl__label">{{ label }}</div>
+		</div>
+		<input
+			:type="type === SELECTION_CONTROL_TYPE.CHECKBOX ? 'checkbox' : 'radio'"
+			class="selectionControl__checkbox"
+			:value="isSelected"
+			@input="onToggle"
```
Comment (kvas-damian) [inline lib/js/components/Form/SelectionControl/SelectionControl.vue:26]: why did we change it?

**Review by kvas-damian — APPROVED** [review]

**Pull request merged**